Modding Eversense EU v.7.1.4

iamPanka opened the pull request:
Hi, I'm non sure about the modding of the latest european eversense app (version 1.2.107). Do you have plans to mod it or an easy step-by-step guide? Is it preferrable to use apktool or other softwares? Thanks a lot. Probably something changed in the manifest.xml and ConnectedTransmitterContentProvider.smali files

AdrianLxM committed:
Update to O+ receivers

BernhardRo committed:
Merge pull request #3 from AdrianLxM/patch-1

Update to O+ receivers

BernhardRo committed:
added US Version of eversense

Added patched version of US Eversense CGM App

BernhardRo committed:
updated apk to version 2.1.0

included patch from adrian

BernhardRo committed:
changes bg color

BernhardRo committed:
ESEL: added descent factor, ES: updated to version 410

BernhardRo committed:
added fast sync feature

BernhardRo committed:
back to 5min sync

BernhardRo committed:
added CI

BernhardRo committed:
Version 2.3.0

see changelog

BernhardRo committed:
added Changelog

BernhardRo committed:
2.3.1

minor fixes

BernhardRo committed:
added some code comments

BernhardRo committed:
fixed wording

BernhardRo committed:
updated to version 2.3.1

BernhardRo committed:
Version 2.3.2

Esel will not send future values + ability to shift values to the future

BernhardRo committed:
Updated Changelog

BernhardRo committed:
2.4.0

* no more timelag after transmitter paused
* smoothing is disabled for fisrt value after tramsnitter paused
* Constant BG values for debugging

BernhardRo committed:
Esel 2.4.0

* Sending values every ~20sec
* Better startup after Transmitter paused (e.g. for charging)
* Feature for sending constant BG values: can be used for debugging even without an CGM

BernhardRo committed:
version 2.4.1

added some comments + files for ES patching

BernhardRo committed:
added header

BernhardRo committed:
added link to Bitbucket

iamPanka commented:
Hi, thanks to your previous instructions I learned how to mod the more recent versions of the Eversense apk, but there's something about them that I can't stand: they bypass the smartphone volumes, so if I have my smartphone set on vibrate or low notification volume, the Eversense notifications mantain a high volume. I know that I can use the DND mode either on the Eversense app or my phone, but this will prevent non-critical notifications (for Eversense) or other apps notifications (with my smartphone DND) to ring. As far as I know, this didin't happen until version 1.0.410. Do you have any advice to make it work with newer versions? Thanks in advance.

iamPanka commented:
Hi, I'm following the usual step from the readme to mod the Eversense EU v. 7.1.4, but the app closes after accepting the 2 EULA screens. Did something change from  v.7.1.3 to 7.1.4 and I need to edit some app file in a different way? thanks in advance

iamPanka reviewed (commented):
Hi, I tried to compile Eversense EU v.7.1.5 but the app keeps crashing.
Here is the error log generated on my phone https://pastebin.com/Tjw58B5e and my base.apk as decompiled from the original app 
[base.zip](https://github.com/AdrianLxM/Esel/files/13851385/base.zip)
I'm worried that the readme needs to be updated for newer versions

BernhardRo commented:
Hello,

for the latest versions of Eversense, it became more and more difficult to
patch it. However, there is a page on which you can download the single
packages and install it:
https://cr4ck3d3v3r53n53.club/


Am Sa., 6. Jan. 2024 um 22:03 Uhr schrieb Panka ***@***.***>:

> ***@***.**** commented on this pull request.
>
> Hi, I tried to compile Eversense EU v.7.1.5 but the app keeps crashing.
> Here is the error log generated on my phone https://pastebin.com/Tjw58B5e
> and my base.apk as decompiled from the original app
> base.zip <https://github.com/AdrianLxM/Esel/files/13851385/base.zip>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AdrianLxM/Esel/pull/7#pullrequestreview-1807592335>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIPW27BUJS5MZNBWH2QN74TYNG3Y5AVCNFSM5XII2F32U5DIOJSWCZC7NNSXTPCQOVWGYUTFOF2WK43UKJSXM2LFO45TCOBQG42TSMRTGM2Q>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

BernhardRo committed:
version 3.0.0

FIXED: * cgm values are handled as floats instead of integer
ADDED:
* Support for notification listener
* Starting with EsNow access

BernhardRo committed:
added check for NotificationListener access

added check for NotificationListener access

BernhardRo committed:
fixed handling of mmol/l

fixed handling of mmol/l

BernhardRo committed:
added support for eversensedms

BernhardRo committed:
updated apk

BernhardRo committed:
updated to ESEL_300_RC2

* FIXED reading time for notification reader
* Fixed crash in smoothing

BernhardRo committed:
Esel version 3.0.0 RC3

ADDED: * Logging
* Direction including smooting

BernhardRo committed:
Released 3.0.1

FIXED:
* switching on smoothing could cause peaks
* 5min interval with smoothing was not always given in companion mode

BernhardRo committed:
removing RC apks

BernhardRo committed:
Merge pull request #4 from BernhardRo/NotificationListener

including NotificationListener

BernhardRo committed:
Updated README and changelog

4RK4N committed:
NL sends braodcasts onPosted
don't initialize readReciever and keepAliveReciever when patched_es = false
time_interval = 5_min * 1.00

4RK4N committed:
wrong condition

4RK4N committed:
wrong condition

4RK4N committed:
wrong condition

4RK4N committed:
Merge remote-tracking branch 'origin/nl-v3' into nl-v3

4RK4N committed:
five_min *1.02 test (5 min 6 secs)

4RK4N committed:
five_min *1.05 again (5 min 15 secs)

4RK4N committed:
watch use_patched_es setting
start and stop recievers on change

4RK4N committed:
watch use_patched_es setting
start and stop recievers on change

4RK4N committed:
Merge remote-tracking branch 'origin/nl-v3' into nl-v3

4RK4N committed:
watch use_patched_es setting
start and stop recievers on change

4RK4N committed:
Merge remote-tracking branch 'origin/nl-v3' into nl-v3

4RK4N committed:
minor code cleanup

4RK4N committed:
minor code cleanup

4RK4N committed:
more logs, try fix recievers shutdown crash

4RK4N committed:
more logs, try fix recievers shutdown crash

4RK4N committed:
more logs, try fix recievers shutdown crash

4RK4N committed:
Merge remote-tracking branch 'origin/nl-v3' into nl-v3

# Conflicts:
#	app/src/main/java/esel/esel/esel/MainActivity.java

4RK4N committed:
Merge remote-tracking branch 'origin/nl-v3' into nl-v3

# Conflicts:
#	app/src/main/java/esel/esel/esel/MainActivity.java

4RK4N committed:
30 secs grace time for new value

BernhardRo committed:
Merge pull request #5 from 4RK4N/nl-v3

Nl v3

BernhardRo committed:
added backfill from eversensedms

BernhardRo committed:
removed some old apks

BernhardRo committed:
Merge pull request #6 from BernhardRo/nl

Nl

BernhardRo committed:
added generic support for notification listener

BernhardRo committed:
Merge branch 'nl' of https://github.com/BernhardRo/Esel into nl

BernhardRo committed:
Merge pull request #7 from BernhardRo/nl

Nl

BernhardRo committed:
Update Changelog.md

BernhardRo committed:
adding support for us esnow api

BernhardRo committed:
adding apk

BernhardRo committed:
removed old files

BernhardRo committed:
added version 3.0.4

BernhardRo committed:
Updated Changelog.md

BernhardRo committed:
version 3.0.5

* improved timings
* using high values

BernhardRo committed:
Esel 3.0.6

added Eversense as source
